### PR TITLE
Correct Lockdown Mode compile guards to support testing

### PR DESCRIPTION
**Pull request by brentfulgham**
#### 0f236a88c9ed99679cb26a8d19483f9445acc61d
<pre>
Correct Lockdown Mode compile guards to support testing
<a href="https://bugs.webkit.org/show_bug.cgi?id=242867">https://bugs.webkit.org/show_bug.cgi?id=242867</a>

Reviewed by Chris Dumez.

* Source/WTF/wtf/PlatformEnableCocoa.h:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(adjustCoreGraphicsForCaptivePortal):

Canonical link: <a href="https://commits.webkit.org/252586@main">https://commits.webkit.org/252586@main</a>
</pre>

### CODE REVIEW

File: Source/WTF/wtf/PlatformEnableCocoa.h [hunk 1]
```diff
@@ -810,8 +810,6 @@
 #define ENABLE_FULLSCREEN_DISMISSAL_GESTURES 1
 #endif
 
-#if (!defined(ENABLE_LOCKDOWN_MODE_API) && \
-    (PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 130000 \
-    || PLATFORM(IOS) && __IPHONE_OS_VERSION_MIN_REQUIRED >= 160000))
+#if (!defined(ENABLE_LOCKDOWN_MODE_API) && (PLATFORM(MAC) || PLATFORM(IOS)))
```
Comment (cdumez) [inline Source/WTF/wtf/PlatformEnableCocoa.h:813]: Should this be PLATFORM(COCOA) ?

**Review by cdumez — APPROVED** [review]
OK if it builds on EWS.

File: Source/WTF/wtf/PlatformEnableCocoa.h (hunk 1 above)
Comment (brentfulgham) [reply Source/WTF/wtf/PlatformEnableCocoa.h:813]: No -- we don't want this to build on WATCHOS, TVOS, or MACCATALYST.

**Comment by brentfulgham** [comment]
Found a missing include when building for iOS Simulator.

**Pull request merged**